src/quote/flashmint/wrapped/provider.ts

```diff
@@ -0,0 +1,106 @@
+import { BigNumber } from '@ethersproject/bignumber'
+
+import {
+  ComponentSwapData,
+  ComponentWrapData,
+  getFlashMintWrappedContract,
+  getIssuanceComponentSwapData,
+  getRedemptionComponentSwapData,
+  getWrapData,
+  slippageAdjustedTokenAmount,
+} from 'utils'
+import { getRpcProvider } from 'utils/rpc-provider'
+
+import { QuoteProvider, QuoteToken } from '../../interfaces'
+
+export interface FlashMintWrappedQuoteRequest {
+  isMinting: boolean
+  inputToken: QuoteToken
+  outputToken: QuoteToken
+  indexTokenAmount: BigNumber
+  slippage: number
+}
+
+export interface FlashMintWrappedQuote {
+  componentSwapData: ComponentSwapData[]
+  componentWrapData: ComponentWrapData[]
+  indexTokenAmount: BigNumber
+  inputOutputTokenAmount: BigNumber
+}
+
+export class WrappedQuoteProvider
+  implements QuoteProvider<FlashMintWrappedQuoteRequest, FlashMintWrappedQuote>
+{
+  constructor(private readonly rpcUrl: string) {}
+
+  async getQuote(
+    request: FlashMintWrappedQuoteRequest
+  ): Promise<FlashMintWrappedQuote | null> {
+    const { inputToken, indexTokenAmount, isMinting, outputToken, slippage } =
+      request
+    console.log(
+      inputToken.symbol,
+      outputToken.symbol,
+      indexTokenAmount.toString(),
+      isMinting,
+      slippage
+    )
+    const indexToken = isMinting ? outputToken : inputToken
+    const indexTokenSymbol = indexToken.symbol
+    const componentSwapData = isMinting
+      ? await getIssuanceComponentSwapData(
+          {
+            indexTokenSymbol,
+            indexToken: indexToken.address,
+            inputToken: inputToken.address,
+            indexTokenAmount,
+          },
+          this.rpcUrl
+        )
+      : await getRedemptionComponentSwapData(
+          {
+            indexTokenSymbol,
+            indexToken: indexToken.address,
+            outputToken: outputToken.address,
+            indexTokenAmount,
+          },
+          this.rpcUrl
+        )
+    console.log(componentSwapData)
+    const componentWrapData = getWrapData(indexToken.symbol)
+    // FIXME: add check
+    // if (componentSwapData.length !== componentWrapData.length) return null
```

Comment:
@pblivin0x with USDCY this will be still a valid check? wrap data length should always equal the amount of the components right?

Reply:
yes thats correct, null data will be passed if no swap/wrap is necessary https://github.com/IndexCoop/index-coop-smart-contracts/blob/509df422a186a73222ba0e3099760093471f5c34/test/integration/ethereum/flashMintWrappedRebasing.spec.ts#L252-L326